server/models/video/video-playlist-element.ts

```diff
@@ -336,7 +336,9 @@ export class VideoPlaylistElementModel extends Model<Partial<AttributesOnly<Vide
     // Internal video?
     if (video.privacy === VideoPrivacy.INTERNAL && accountId) return VideoPlaylistElementType.REGULAR
 
-    if (video.privacy === VideoPrivacy.PRIVATE || video.privacy === VideoPrivacy.INTERNAL) return VideoPlaylistElementType.PRIVATE
+    if (new Set([ VideoPrivacy.PRIVATE, VideoPrivacy.INTERNAL, VideoPrivacy.PASSWORD_PROTECTED ]).has(video.privacy)) {
```

Comment:
I will add a comment. However this will have no impact on the owner, as we are dealing with the case of being the owner just above :

https://github.com/Chocobozzz/PeerTube/blob/5d015d9e6cfef6efe6978f8e0e4124cdbcc33d92/server/models/video/video-playlist-element.ts#L334

Reply:
Oh nice!